src/settlement-chain/PayerReportManager.sol

```diff
@@ -0,0 +1,381 @@
+     * @param  payerReportsByOriginator The mapping of arrays of payer reports by originator node ID.
+     */
+    struct PayerReportManagerStorage {
+        mapping(uint32 originatorId => PayerReport[] payerReports) payerReportsByOriginator;
+    }
+
+    // keccak256(abi.encode(uint256(keccak256("xmtp.storage.PayerReportManager")) - 1)) & ~bytes32(uint256(0xff))
+    bytes32 internal constant _PAYER_REPORT_MANAGER_STORAGE_LOCATION =
+        0x26b057ee8e4d60685198828fdf1c618ab8e36b0ab85f54a47b18319f6f718e00;
+
+    function _getPayerReportManagerStorage() internal pure returns (PayerReportManagerStorage storage $) {
+        // slither-disable-next-line assembly
+        assembly {
+            $.slot := _PAYER_REPORT_MANAGER_STORAGE_LOCATION
+        }
+    }
+
+    /* ============ Constructor ============ */
+
+    /**
+     * @notice Constructor for the implementation contract, such that the implementation cannot be initialized.
+     * @param  parameterRegistry_ The address of the parameter registry.
+     * @param  nodeRegistry_      The address of the node registry.
+     * @param  payerRegistry_     The address of the payer registry.
+     * @dev    The parameter registry, node registry, and payer registry must not be the zero address.
+     * @dev    The parameter registry, node registry, and payer registry are immutable so that they are inlined in the
+     *         contract code, and have minimal gas cost.
+     */
+    constructor(address parameterRegistry_, address nodeRegistry_, address payerRegistry_) ERC5267() {
+        require(_isNotZero(parameterRegistry = parameterRegistry_), ZeroParameterRegistry());
+        require(_isNotZero(nodeRegistry = nodeRegistry_), ZeroNodeRegistry());
+        require(_isNotZero(payerRegistry = payerRegistry_), ZeroPayerRegistry());
+
+        _disableInitializers();
+    }
+
+    /* ============ Initialization ============ */
+
+    /// @inheritdoc IPayerReportManager
+    function initialize() public initializer {
+        _initializeERC5267();
+    }
+
+    /* ============ Interactive Functions ============ */
+
+    /// @inheritdoc IPayerReportManager
+    function submit(
+        uint32 originatorNodeId_,
+        uint32 attesterNodeId_,
+        uint32 startSequenceId_,
+        uint32 endSequenceId_,
+        bytes32 payersMerkleRoot_,
+        address[] calldata nodeOperators_,
+        PayerReportSignature[] calldata signatures_
+    ) external returns (uint256 payerReportIndex_) {
+        PayerReport[] storage payerReports_ = _getPayerReportManagerStorage().payerReportsByOriginator[
+            originatorNodeId_
+        ];
+
+        payerReportIndex_ = payerReports_.length;
+
+        uint32 lastSequenceId_ = payerReportIndex_ > 0 ? payerReports_[payerReportIndex_ - 1].endSequenceId : 0;
+
+        // Enforces that the start sequence ID is the next expected sequence ID.
+        if (startSequenceId_ != lastSequenceId_ + 1) {
```

Comment:
The way we are handling this on the nodes is that the `startSequenceID` is _equal_ to the end sequence ID of the last report. That way we can handle the rare, but possible, case where an originator is missing a sequence ID one above the end of the last report.

Reply:
So that would imply that a Payer Report is inclusive of `startSequenceID` and exclusive of its `endSequenceID`. Is that correct?